Adding Swagger to the project

jegor-chomczuk opened the pull request.

jegorchomczuk committed:
Adding Swagger to the project

Gieted commented on build.gradle.kts:
```diff
@@ -38,6 +38,9 @@ dependencies {
     implementation("org.jetbrains.kotlin:kotlin-stdlib:1.6.21")
 
+    implementation("io.springfox:springfox-boot-starter:3.0.0")
+    implementation("com.google.guava:guava:30.0-android")
```
Czemu guava-android? Chyba jakaś pomyłka

Gieted commented on src/test/java/com/workspaceservice/WorkspaceControllerTest.java:
```diff
@@ -90,25 +92,25 @@ void createWorkspace() throws Exception {
+
+        assertTrue(result.getResponse().getStatus() == 201);
+
+        var path = "C:\\Users\\ykhom\\IdeaProjects\\workspace-service\\test_repos\\" + workspaceId + "\\" + addFilesList.get(1).getPath();
```
Nie używaj ścieżek absolutnych, na innym komputerze ten test nie zadziała

Gieted commented on src/test/java/com/workspaceservice/WorkspaceControllerTest.java:
```diff
@@ -60,8 +60,10 @@ void tearDown() {
-        MvcResult result = mockMvc.perform(get("/workspace/all"))
-                .andExpect(status().isOk()).andReturn();
+        MvcResult result = mockMvc
+                .perform(get("/workspace/all"))
```
Moglibyśmy zmienić nazwę tego endpointu na `/workspaces`? Byłoby bardziej restowo

jegorchomczuk committed:
1. Proper guava lib added.
2. Endpoint names fixet from /workspace/all to /workspaces
3. Absolute path removed in WorkspaceControllerTest.java

Gieted commented on src/test/java/com/workspaceservice/WorkspaceControllerTest.java:
```diff
@@ -90,29 +95,29 @@ void createWorkspace() throws Exception {
+
+        assertTrue(result.getResponse().getStatus() == 201);
+
+        var path = "test_repos\\" + workspaceId + "\\" + addFilesList.get(1).getPath();
```
Use forward slashes instead of backslashes

jegorchomczuk committed:
Forward slashes instead of backslashes

Gieted committed:
Merge branch 'main' into swagger

# Conflicts:
#	src/main/java/com/workspaceservice/controller/WorkspaceController.java
